hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRequest.java

```diff
@@ -661,9 +661,17 @@ private OmKeyInfo prepareMultipartFileInfo(
     // error no such multipart upload.
     String uploadID = args.getMultipartUploadID();
     Preconditions.checkNotNull(uploadID);
-    String multipartKey = omMetadataManager
-            .getMultipartKey(args.getVolumeName(), args.getBucketName(),
-                    args.getKeyName(), uploadID);
+    String multipartKey = "";
+    if (omPathInfo != null) {
+      // FileTable metadata format
+      multipartKey =
+              omMetadataManager.getMultipartKey(omPathInfo.getLastKnownParentId(),
+                      omPathInfo.getLeafNodeName(), uploadID);
+    } else {
+      multipartKey = omMetadataManager
```

Comment:
This code should ideally never be called, as this function is specific to Prefix optimizations. Should be add a precondition here saying, Not reachable code ?

Reply:
Thanks @mukul1987 for the review comments.

`OMKeyRequest#prepareMultipartFileInfo()` is a common function, used by both V0 and V1 code. I've used `omPathInfo != null` as a differentiator between V0 and V1.

V0 code flow:-
`OMKeyRequest#prepareKeyInfo --> OMKeyRequest#prepareFileInfo --> OMKeyRequest#prepareMultipartFileInfo()`